### PR TITLE
Support null locale before locale initialization, Roll engine to cdd592f5be97e824

### PR DESCRIPTION
**Pull request by GaryQian**
The engine will roll a change that makes uninitialized locales null very soon, but this still works without the change.

Engine Roll:
git log --oneline --no-merges 4c79e423dc6f89f98d8ceb263a5ca78e2f2da996..cdd592f5be97e8245b31d7621bc95234b62c3a43
cdd592f5b Reland 'Pass null instead of 'none' locale' (#6632)
2cd89200c Roll src/third_party/skia b1a002e850e1..327955b1ba19 (12 commits) (#6631)
edfe02481 13771 - iOS dictation bug (#6607)
cadf4405b Roll src/third_party/skia b9998cdceec7..b1a002e850e1 (13 commits) (#6626)

### CODE REVIEW

File: packages/flutter_localizations/test/widgets_test.dart [hunk 1]
```diff
@@ -581,7 +581,8 @@ void main() {
       )
     );
 
-    // Initial WidgetTester locale is new Locale('', '')
+    // Initial WidgetTester locale is null due to no platform intitializing it.
+    await tester.binding.setLocale('', '');
```
Comment (tvolkert) [inline packages/flutter_localizations/test/widgets_test.dart]: What happens if we don't do this?  Can we adjust the expectations below?  Namely, it seems like `$locale` shouldn't return "_", so we should probably adjust the `toString()` implementation of `Locale`

Comment (GaryQian) [reply packages/flutter_localizations/test/widgets_test.dart]: Without this, the null handling (after the engine-side change) resolves the locale to the first supported locale, which is en_US. I'll go ahead and check for en_US and roll the engine with this test change.

Comment (GaryQian) [reply packages/flutter_localizations/test/widgets_test.dart]: There should be no more Locale("", "") anymore, at least not produced by flutter.

**Review by tvolkert — APPROVED** [review]

**Review by yjbanov — APPROVED** [review]

**Comment by yjbanov** [comment]
lgtm

**Review by tvolkert — APPROVED** [review]

**Comment by GaryQian** [comment]
Closing and reopening to attempt to force in latest changes (github is not updating this PR)

**Pull request closed without merging**